cli/internal/globwatcher/globwatcher.go

```diff
@@ -135,24 +146,58 @@ func (g *GlobWatcher) OnFileWatchEvent(ev filewatcher.Event) {
 			g.logger.Error(fmt.Sprintf("failed to check path %v against glob %v: %v", repoRelativePath, glob, err))
 			continue
 		}
-		// If this glob matches, we know that it has changed for every hash that included this glob.
+		// If this glob matches, we know that it has changed for every hash that included this glob
+		// and is not excluded by a hash's exclusion globs.
 		// So, we can delete this glob from every hash tracking it as well as stop watching this glob.
 		// To stop watching, we unref each of the directories corresponding to this glob.
 		if matches {
-			delete(g.globStatus, glob)
+			// We need to check if at least one hash invalidates this glob. If that is true,
+			// we remove the glob from globStatus.
+			globInvalidatedByHash := false
```

Comment:
Reading this comment, this doesn't seem right. I *think* what we want is that if _all_ hashes don't exclude this change, we can invalidate the glob and remove it. But, if at least one hash excludes this change, then as far as we know, this glob is still unchanged for that hash.

Reply:
I think we probably need a test case with multiple hashes watching the same glob with different sets of exclusions

Reply:
Yeah...realized I wasn't thinking this through. Rewrote it to just remove the specific hash that is invalidated from `globStatus[glob]` and then if we remove everything, delete the entry then.